src/Services/FhirResourcesService.php

```diff
@@ -152,6 +152,39 @@ public function createEncounterResource($eid = '', $data = '', $encode = true)
             return $resource;
         }
     }
+	
+	public function parsePatientResource($fhirJson)
```

Comment:
My comment wasn't showing so i'll try again.
Try using the `public function parseResource($rjson = '', $scheme = 'json')`
This function will know what resource is being parsed and parse and populate the appropriate resource class and return the json as php array if you wanted to instantiate class yourself and pass the array to it. 
Then call properties with class methods. This way when changing fhir versions then the classes will handle locating the properties and we won't have to modify a bunch of hard coded json's.

This is still okay if you want me to bring in and i'll modify as an example for you and others to follow.

Reply:
> This is still okay if you want me to bring in and i'll modify as an example for you and others to follow.

Yes, that would be great. The example would help in the implementation of the further resources.